be/src/storage/tablet_manager.cpp

```diff
@@ -906,55 +911,21 @@ Status TabletManager::start_trash_sweep() {
                            << ". tablet_uid=" << tablet->tablet_uid();
                 continue;
             }
-            // move data to trash
-            auto tablet_id_path = tablet->tablet_id_path();
-            st = Env::Default()->path_exists(tablet_id_path);
-            if (st.ok() && config::trash_file_expire_time_sec > 0) {
-                // take snapshot of tablet meta for recovery
-                if (tablet->keys_type() == KeysType::PRIMARY_KEYS) {
-                    if (st = SnapshotManager::instance()->make_snapshot_on_tablet_meta(tablet); !st.ok()) {
-                        LOG(WARNING) << "Fail to make_snapshot_on_tablet_meta, tablet_id=" << tablet->tablet_id()
-                                     << " schema_hash=" << tablet->schema_hash() << ", status=" << st.to_string();
-                        continue;
-                    }
-                } else {
-                    auto meta_file_path = fmt::format("{}/{}.hdr", tablet->schema_hash_path(), tablet->tablet_id());
-                    tablet->tablet_meta()->save(meta_file_path);
-                }
-                if (st = move_to_trash(tablet_id_path); st.ok()) {
-                    LOG(INFO) << "Moved " << tablet_id_path << " to trash";
-                } else {
-                    LOG(WARNING) << "Fail to move " << tablet_id_path << " to trash";
-                    continue;
-                }
-            } else if (st.ok()) {
-                // Ignore the result.
-                (void)FileUtils::remove_all(tablet_id_path);
-            } else if (st.is_not_found()) {
-                st = Status::OK();
-            } else {
-                LOG(WARNING) << "Fail to check " << tablet_id_path << ": " << st;
+            // remove tablet meta
+            if (st = _remove_tablet_meta(tablet); !st.ok()) {
```

Comment:
If tablet_meta is already removed, the trash will missed the meta.

Reply:
Yes, you're right, I'll fix it later.

Reply:
fixed